storm-core/src/jvm/backtype/storm/utils/ShellProcess.java

```diff
@@ -48,8 +50,31 @@ public ShellProcess(String[] command) {
         this.command = command;
     }
 
+    public void setEnv(Map<String, String> env) {
+        this.env = env;
+    }
+
+    private void modifyEnvironment(Map<String, String> buildEnv) {
+        for (Map.Entry<String, String> entry : env.entrySet()) {
+            if ("PATH".equals(entry.getKey())) {
+                buildEnv.put("PATH", buildEnv.get("PATH") + File.pathSeparatorChar + env.get("PATH"));
+            } else {
+                buildEnv.put(entry.getKey(), entry.getValue());
+            }
+        }
+    }
+
+
     public Number launch(Map conf, TopologyContext context) {
         ProcessBuilder builder = new ProcessBuilder(command);
+        if (!env.isEmpty()) {
+            Map<String, String> buildEnv = builder.environment();
+            modifyEnvironment(buildEnv);
+        }
+        for(Map.Entry<String, String> entry : builder.environment().entrySet()) {
```

Comment:
@HeartSaVioR , Sorry I am not sure what you meant here. The code (line 70) is checking if the user had passed any custom environment map only after which it calls the modifyEnvironment with the existing environment map for modification.  Dd you mean to include debug message for better clarity within this block ?